packages/gatsby/src/db/nodes.js

```diff
@@ -1,8 +1,21 @@
+/* @flow */
 const _ = require(`lodash`)
 const { store } = require(`../redux`)
+const { run: runQuery } = require(`./nodes-query`)
+
+interface NodeStore {
+  getNodes: () => Array<any>;
+  getNode: (id: string) => any | undefined;
+  getNodesByType: (type: string) => Array<any>;
+  getTypes: () => Array<string>;
+  hasNodeChanged: (id: string, digest: string) => boolean;
+  getNodeAndSavePathDependency: (id: string, path: string) => any | undefined;
+  // XXX(freiksenet): types
+  runQuery: (...args: any) => any | undefined;
```

Comment:
We can use the current runQuery, but we should think about extracting some of the changes I did, e.g. resolving nodes for sort, group, and distinct fields; also not relevant right now but when we add addTypeDefs: handling NonNull fields

Reply:
Also: when manually calling resolvers in [`run-sift`](https://github.com/gatsbyjs/gatsby/blob/9168d1f912017d3078decd7ea72f7a123ccbdbe3/packages/gatsby/src/redux/run-sift.js#L104), we should put the nodeModel on context as well

Reply:
And: if we use something like `getValueAtSelector` instead of `_.get` for [sorting in `run-sift`](https://github.com/gatsbyjs/gatsby/blob/9168d1f912017d3078decd7ea72f7a123ccbdbe3/packages/gatsby/src/redux/run-sift.js#L256), we can sort on properties on objects in arrays - this would be either a bugfix or breaking change

Reply:
We should extract the changes that resolve fields so that we can have unified logic for it. Ideally stuff in what we call "nodeStorage" shouldn't care about GraphQL Types. But let's do it after releasing this.

Reply:
Sounds good. I'll remove the resolving for sort fields that I apparently sneaked in (oops).